rclcpp/src/rclcpp/duration.cpp

```diff
@@ -67,10 +67,18 @@ Duration::operator builtin_interfaces::msg::Duration() const
 {
   builtin_interfaces::msg::Duration msg_duration;
   constexpr rcl_duration_value_t kDivisor = RCL_S_TO_NS(1);
+  constexpr std::int32_t limit_s = std::numeric_limits<std::int32_t>::max();
+  constexpr std::uint32_t limit_ns = std::numeric_limits<std::uint32_t>::max();
   const auto result = std::div(rcl_duration_.nanoseconds, kDivisor);
   if (result.rem >= 0) {
-    msg_duration.sec = static_cast<std::int32_t>(result.quot);
-    msg_duration.nanosec = static_cast<std::uint32_t>(result.rem);
+    // saturate if we will overflow
+    if (result.quot > limit_s) {
```

Comment:
should we check if `quot < min_limit` as well?
We're almost not using negative durations, but IIUC those are valid

Reply:
Yes, it's probably good to cover the negative case too. see e4636b1